src/adapters/EthereumGeneralAdapter1.sol

```diff
@@ -48,18 +52,35 @@ contract EthereumGeneralAdapter1 is GeneralAdapter1 {
 
         ST_ETH = IWstEth(wStEth).stETH();
         WST_ETH = wStEth;
-        MORPHO_TOKEN = morphoToken;
         MORPHO_WRAPPER = morphoWrapper;
+        MORPHO_TOKEN = morphoToken;
+        MORPHO_TOKEN_LEGACY = address(ERC20Wrapper(morphoWrapper).underlying());
     }
 
     /* MORPHO TOKEN WRAPPER ACTIONS */
 
+    /// @notice Wraps Morpho tokens.
+    /// @dev Legacy Morpho tokens must have been previously sent to the adapter.
+    /// @param receiver The address to send the tokens to.
+    /// @param amount The amount of tokens to wrap. Pass `type(uint).max` to wrap the adapter's balance of legacy Morpho
+    /// tokens.
+    function morphoWrapperDepositFor(address receiver, uint256 amount) external onlyBundler3 {
+        // Do not check `receiver` against the zero address as it's done at the Morpho Wrapper's level.
+        if (amount == type(uint256).max) amount = IERC20(MORPHO_TOKEN_LEGACY).balanceOf(address(this));
+
+        require(amount != 0, ErrorsLib.ZeroAmount());
+
+        // The MORPHO wrapper's allowance is not reset as it is trusted.
+        SafeERC20.forceApprove(IERC20(MORPHO_TOKEN_LEGACY), MORPHO_WRAPPER, type(uint256).max);
+
+        require(ERC20Wrapper(MORPHO_WRAPPER).depositFor(receiver, amount), ErrorsLib.DepositFailed());
```

Comment:
fine according to [this](https://github.com/morpho-org/morpho-token/blob/d828ef0559a6b8ef52d3901e4e76286e1b429900/src/Wrapper.sol#L48), though I'm not convinced it's a good idea
```suggestion
        ERC20Wrapper(MORPHO_WRAPPER).depositFor(receiver, amount);
```